addon/utils/rdfa/event-processor.ts

```diff
@@ -128,24 +114,35 @@ export default EmberObject.extend({
    * @public
    */
   handleRegistryChange(/*registry*/) {
-    const editor = this.get('editor');
-    editor.clearHighlightForLocations(this.get('cardsLocationFlaggedRemoved'));
-
-    this.get('cardsLocationFlaggedNew').forEach(location => {
-      editor.highlightRange(location[0], location[1]);
-    });
+    let region;
+    while (region = this.cardsLocationFlaggedRemoved.shift()) {
```

Comment:
Eslint doesn't like this trick, personally I'm indifferent towards it

Reply:
also, pop is O(1) and shift is O(n), so if the order doesn't matter pop is better

Reply:
I'm removing them in the order they were added I think (so first in first out), for the size of the array there will be no difference in O(1) vs O(n). IIRC the difference between shift and pop only shows up on arrays in excess of 100.000 elements.

Reply:
wrt to the assignment in the loop predicate, I'm fine with having a different predicate and doing the assignment inside the loop as well if that makes the linter happy :).